extensions/procfs/tests/ProcFsMonitorTests.cpp

```diff
@@ -0,0 +1,157 @@
+
+  SECTION("Absolute JSON") {
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::ResultRelativenessProperty.getName(), "Absolute");
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::OutputFormatProperty.getName(), "JSON");
+    const auto& result = test_controller_.trigger();
+
+    REQUIRE(result.at(ProcFsMonitor::Success).size() == 1);
+    const auto& result_flow_file = result.at(ProcFsMonitor::Success)[0];
+
+    rapidjson::Document document;
+    auto content = test_controller_.plan->getContent(result_flow_file);
+    document.Parse(content.c_str());
+    REQUIRE(document.IsObject());
+    REQUIRE(document.HasMember("CPU"));
+    CHECK(document["CPU"].HasMember("cpu"));
+    CHECK(document.HasMember("Disk"));
+    CHECK(document.HasMember("Network"));
+    CHECK(document.HasMember("Process"));
+    CHECK(document.HasMember("Memory"));
+  }
+
+  SECTION("Absolute OpenTelemetry")  {
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::ResultRelativenessProperty.getName(), "Absolute");
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::OutputFormatProperty.getName(), "OpenTelemetry");
+    const auto& result = test_controller_.trigger();
+
+    REQUIRE(result.at(ProcFsMonitor::Success).size() == 1);
+    const auto& result_flow_file = result.at(ProcFsMonitor::Success)[0];
+
+    rapidjson::Document document;
+    auto content = test_controller_.plan->getContent(result_flow_file);
+    document.Parse(content.c_str());
+    REQUIRE(document.IsObject());
+    REQUIRE(document.HasMember("Body"));
+    REQUIRE(document["Body"].HasMember("CPU"));
+    CHECK(document["Body"]["CPU"].HasMember("cpu"));
+    CHECK(document["Body"].HasMember("Disk"));
+    CHECK(document["Body"].HasMember("Network"));
+    CHECK(document["Body"].HasMember("Process"));
+    CHECK(document["Body"].HasMember("Memory"));
+  }
+
+  SECTION("Relative JSON") {
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::ResultRelativenessProperty.getName(), "Relative");
+    test_controller_.plan->setProperty(proc_fs_monitor, ProcFsMonitor::OutputFormatProperty.getName(), "JSON");
+    {
+      const auto& result = test_controller_.trigger();
+
+      REQUIRE(result.at(ProcFsMonitor::Success).size() == 1);
+      const auto& result_flow_file = result.at(ProcFsMonitor::Success)[0];
+
+      rapidjson::Document document;
+      auto content = test_controller_.plan->getContent(result_flow_file);
+      document.Parse(content.c_str());
+      REQUIRE(document.IsObject());
+      // First trigger has not enough information for relative output
+      CHECK_FALSE(document.HasMember("CPU"));
+      CHECK_FALSE(document.HasMember("Disk"));
+      CHECK_FALSE(document.HasMember("Network"));
+      CHECK_FALSE(document.HasMember("Process"));
+      CHECK(document.HasMember("Memory"));
+    }
+    sleep(1);
```

Comment:
is this the linux `sleep` syscall? could we use `std::this_thread::sleep_for` here?

Reply:
is this in seconds? could we lower the wait time?

Reply:
good idea, changed it in https://github.com/apache/nifi-minifi-cpp/pull/1152/commits/f06938c35f0943a66c4bd1d7b895a02da8c0a83f